third_party/move/move-bytecode-verifier/src/dependencies.rs

```diff
@@ -332,6 +339,31 @@ fn verify_imported_functions(context: &Context) -> PartialVMResult<()> {
                     owner_module,
                 )
                 .map_err(|e| e.at_index(IndexKind::FunctionHandle, idx as TableIndex))?;
+
+                // Compatible attributes.
+                let mut def_attrs = def_handle.attributes.as_slice();
+                let handle_attrs = function_handle.attributes.as_slice();
+                if !handle_attrs.is_empty() && def_attrs.is_empty() {
+                    // This is a function with no attributes, which can come from that
+                    // it's compiled for < Move 2.2. Synthesize the
+                    // `persistent` attribute from Public visibility, which we find
+                    // in the definition.
+                    if owner_module.function_def_at(*def_def_idx).visibility == Visibility::Public {
+                        def_attrs = &[FunctionAttribute::Persistent]
```

Comment:
This looks like a ticking bomb to me: you reset the def attributes to persistent only to do the check, but if someone adds more attributes later to the check, things break. You should instead use the definition attributes union persistent?

Reply:
This is guarded by `def_atttrs.is_empty()`, so I think its fine.